x-pack/plugins/security/server/authentication/api_keys/api_keys.ts

```diff
@@ -175,20 +178,39 @@ export class APIKeys {
    */
   async create(
     request: KibanaRequest,
-    params: CreateAPIKeyParams
+    {
+      kibana_feature_descriptors: kibanaFeatureDescriptors,
+      role_descriptors: roleDescriptors,
+      name,
+      metadata,
+      expiration,
+    }: CreateAPIKeyParams
   ): Promise<CreateAPIKeyResult | null> {
     if (!this.license.isEnabled()) {
       return null;
     }
 
+    const { applications } = transformPutPayloadToElasticsearchRole(
+      { kibana: kibanaFeatureDescriptors, elasticsearch: {} },
+      'kibana-.kibana'
+    );
+
     this.logger.debug('Trying to create an API key');
 
+    if (applications.length > 0) {
+      Object.keys(roleDescriptors).forEach((role) => {
+        const roleDescriptor = roleDescriptors[role];
+
+        roleDescriptor.applications = applications;
+      });
+    }
```

Comment:
Since `role_descriptors` and `kibana_feature_descriptors` should be mutually exclusive, we shouldn't be using one to augment the other